packages/crons/src/close-governance.ts

```diff
@@ -114,6 +128,65 @@ import pLimit from "p-limit";
     );
     await batchParallelInstructionsWithPriorityFee(provider, relinquishIxns);
 
+    const proxyAssignments = await proxyProgram.account.proxyAssignmentV0.all();
+    const expiredProxyAssignments = proxyAssignments.filter((pa) => {
+      const { account } = pa;
+      return account.expirationTime.lt(new anchor.BN(Number(solanaTime)));
+    });
+
+    const proxyAssignmentsByAsset: {
+      [key: string]: (typeof proxyAssignments)[0][];
+    } = expiredProxyAssignments.reduce(
+      (acc, assignment) => ({
+        ...acc,
+        [assignment.account.asset.toBase58()]: [
+          ...(acc[assignment.account.asset.toBase58()] || []),
+          assignment,
+        ],
+      }),
+      {}
+    );
+
+    const multiDimArray: TransactionInstruction[][] = await Promise.all(
+      Object.entries(proxyAssignmentsByAsset).map(async ([_, proxies], idx) => {
+        const sortedProxies = proxies.sort((a, b) =>
+          a.account.index < b.account.index ? 1 : -1
+        );
+
+        return (
+          await Promise.all(
+            sortedProxies.map(async (proxy, index) => {
+              if (index === sortedProxies.length - 1) {
+                return proxyProgram.methods
+                  .closeExpiredProxyV0()
+                  .accounts({
+                    proxyAssignment: new PublicKey(proxy.publicKey),
+                  })
+                  .instruction();
+              }
+
+              const prevProxyAssignment = new PublicKey(
+                sortedProxies[index + 1].publicKey
+              );
+
+              return proxyProgram.methods
+                .unassignExpiredProxyV0()
+                .accounts({
+                  prevProxyAssignment,
+                  proxyAssignment: new PublicKey(proxy.publicKey),
+                })
+                .instruction();
+            })
+          )
+        ).filter(truthy);
+      })
+    );
+
+    await batchSequentialParallelInstructions({
```

Comment:
Don't we have a helper to batch instructions to txs packing as tightly as possible?

Reply:
does that then run them in sequential order or parallel?